### ISSUE TITLE
Greycoprops correlation always return 1 if greycomatrix is not normalized

### ISSUE THREAD

**Issue by romuere** [scikit-image/scikit-image#2012]
```
img = (np.random.random((100,100))*8).astype(np.uint8)
p = feature.greycomatrix(img,[1,2,4,5],[0, 0.25, 1, 1.5],levels=8)
feature.greycoprops(p,'correlation')
```

**Comment by sdsingh** [comment]
The angles specified seem a bit odd--they are intended to be in radians, as in the example, so this may cause some errors. Not sure what a pixel window of 2, in direction 0.25 radians would do. 

However, it is still odd that the correlation is 1 for the 0 angle case.

**Comment by sdsingh** [comment]
Took a deeper look. If the flag "normed=True" is passed, the results are reasonable (seemingly correct). 

Without the normed value, the standard deviations in the GLCM overflow, and everything gets clipped to 1 in the correlation.

**Comment by sdsingh** [comment]
I think we should consider changing the GLCM API to an offset model, like MATLAB, where x and y offsets are given. Or, if this is opposed, I guess we can just improve the documentation to clarify the meaning of the angles.

**Comment by sdsingh** [comment]
In addition, now, it is clear to me that the GLCM matrix should be normalized before computing any statistics. I think this should probably be the only resolution to this. 
Reference 1:
[GLCM Paper](http://www.uio.no/studier/emner/matnat/ifi/INF4300/h08/undervisningsmateriale/glcm.pdf)

Reference 2:
http://www.mathworks.com/help/images/ref/graycoprops.html

**Comment by romuere** [comment]
Hi Shaun.

It is true, the GLCM has to be normalized! Thanks for your answer!!

Best regards,


<<---------------------------------------------------------------------->>
Romuere Silva
Bacharel em Ciência da Computação - UFPI
Mestre em Ciência da Computação - UFPI
Doutorando em Engenharia de Teleinformática - UFC
Professor UFPI - CSHNB
http://lattes.cnpq.br/3734588405818556
<<---------------------------------------------------------------------->>

2016-04-13 17:53 GMT-07:00 Shaun Singh <notifications@github.com>:

**Comment by sciunto** [comment]
@sdsingh Would you like to open a pull request?

**Comment by jni** [comment]
Closed by #2532

**Comment by michaelaye** [comment]
I don't understand why the GLCM ALWAYS has to be normalized? If that was true, why was it ever implemented without normalization? I can't believe it was useless before?
